src/main/java/uk/gov/hmcts/reform/civil/model/citizenui/CcdDashboardClaimMatcher.java

```diff
@@ -66,4 +90,149 @@ public boolean hasResponseDeadlineBeenExtended() {
         return caseData.getRespondent1TimeExtensionDate() != null
             && caseData.getCcdState() == CaseState.AWAITING_RESPONDENT_ACKNOWLEDGEMENT;
     }
+
+    @Override
+    public boolean isOrderMade() {
+        return (caseData.getCcdState() == CaseState.All_FINAL_ORDERS_ISSUED
+            || caseData.getCcdState() == CaseState.CASE_PROGRESSION)
+            && getTimeOfLastNonSDOOrder().isPresent();
+    }
+
+    @Override
+    public Optional<LocalDateTime> getTimeOfLastNonSDOOrder() {
+        if (caseData.getCcdState() == CaseState.CASE_PROGRESSION
+            || caseData.getCcdState() == CaseState.All_FINAL_ORDERS_ISSUED) {
+            return getTimeOfMostRecentEventOfType(EnumSet.of(
+                CaseEvent.COURT_OFFICER_ORDER,
+                CaseEvent.GENERATE_DIRECTIONS_ORDER
+            ));
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    protected Optional<LocalDateTime> getSDOTime() {
+        return caseData.getSDODocument().map(d -> d.getValue().getCreatedDatetime());
+    }
+
+    @Override
+    public boolean isSDOOrderCreated() {
+        Optional<LocalDateTime> lastNonSdoOrderTime = getTimeOfLastNonSDOOrder();
+        Optional<LocalDateTime> sdoTime = getSDOTime();
+        return caseData.getHearingDate() == null
+            && CaseState.CASE_PROGRESSION.equals(caseData.getCcdState())
+            && !isSDOOrderLegalAdviserCreated()
+            && !isSDOOrderInReview()
+            && !isSDOOrderInReviewOtherParty()
+            && !isDecisionForReconsiderationMade()
+            && sdoTime.isPresent()
+            && (lastNonSdoOrderTime.isEmpty()
+            || sdoTime.get().isAfter(lastNonSdoOrderTime.get()));
+    }
+
+    @Override
+    public boolean isSDOOrderLegalAdviserCreated() {
+        Optional<LocalDateTime> lastNonSdoOrderTime = getTimeOfLastNonSDOOrder();
+        Optional<LocalDateTime> sdoTime = getSDOTime();
+        return featureToggleService.isCaseProgressionEnabled()
+            && caseData.getHearingDate() == null
+            && isSDOMadeByLegalAdviser()
+            && !isSDOOrderInReview()
+            && !isSDOOrderInReviewOtherParty()
+            && !isDecisionForReconsiderationMade()
+            && sdoTime.isPresent()
+            && (lastNonSdoOrderTime.isEmpty()
+            || sdoTime.get().isAfter(lastNonSdoOrderTime.get()));
+    }
+
+    @Override
+    public boolean isMoreDetailsRequired() {
+        Optional<LocalDateTime> lastOrder = getTimeOfLastNonSDOOrder();
+        Optional<LocalDateTime> sdoTime = getSDOTime();
+        return sdoTime.isPresent()
+            && caseData.getHearingDate() == null
+            && featureToggleService.isCaseProgressionEnabled()
+            && (lastOrder.isEmpty()
+            || lastOrder.get().isBefore(sdoTime.get()));
+    }
+
+    @Override
+    public boolean isHearingLessThanDaysAway(int i) {
```

Comment:
If this goes around again, change this from 'i'.

Reply:
Should be resolved now